src/main.c

```diff
@@ -74,12 +74,42 @@ void chat_add(int channel, unsigned int color, const char* msg) {
 }
 char chat_popup[256] = {};
 int chat_popup_color;
+float chat_popup_height = 0.0F;
 float chat_popup_timer = 0.0F;
 float chat_popup_duration = 0.0F;
 
 void chat_showpopup(const char* msg, float duration, int color) {
 	strcpy(chat_popup, msg);
 	chat_popup_timer = window_time();
+
+	chat_popup_height = font_fit_height(settings.window_width, BIG_TEXT_SIZE, chat_popup);
+
+	int len = strlen(chat_popup);
+	char tmp[len + 1];
+	strcpy(tmp, chat_popup);
+
+	char* tmp_offset = tmp;
+	while(chat_popup_height == -1.0F) {
+		// break up chat_popup into multiple lines
+
+		// find first space from the left of the middle of chat_popup
+		// replace with newline
+		char* pos = tmp_offset + len / 2;
+		while(*pos != ' ' && pos != tmp_offset)
+			pos--;
+
+		if(pos != tmp_offset)
+			*pos = '\n';
+		else {
+			chat_popup_height = 16.0F;
+			break;
+		}
+
+		chat_popup_height = font_fit_height(settings.window_width, BIG_TEXT_SIZE, pos);
```

Comment:
if all else fails, just use whatever height will fit if only the last line is drawn

Reply:
should probably just be a min(chat_popup_height, font_fit_height(settings.window_width, BIG_TEXT_SIZE, pos))